bluesky/callbacks/mpl_plotting.py

```diff
@@ -7,14 +7,242 @@
+
+        # Decide which plots we want for this stream.
+
+        self._descriptors[doc['uid']] = doc
+        stream_name = doc.get('name', 'primary')  # fall back for old docs
+        columns = hinted_fields(doc)
+
+        # ## This deals with old documents. ## #
+
+        if stream_name == 'primary' and self._cleanup_motor_heuristic:
+            # We stashed object names in self.dim_fields, which we now need to
+            # look up the actual fields for.
+            self._cleanup_motor_heuristic = False
+            fixed_dim_fields = []
+            for obj_name in self.dim_fields:
+                # Special case: 'time' can be a dim_field, but it's not an
+                # object name. Just add it directly to the list of fields.
+                if obj_name == 'time':
+                    fixed_dim_fields.append('time')
+                    continue
+                try:
+                    fields = doc.get('hints', {}).get(obj_name, {})['fields']
+                except KeyError:
+                    fields = doc['object_keys'][obj_name]
+                fixed_dim_fields.extend(fields)
+            self.dim_fields = fixed_dim_fields
+
+        # ## DECIDE WHICH KIND OF PLOT CAN BE USED ## #
+
+        if not self._plots_enabled:
+            return
+        if stream_name in self.noplot_streams:
+            return
+        if not columns:
+            return
+        if ((self._start_doc.get('num_points') == 1) and
+                (stream_name == self.dim_stream) and
+                self.omit_single_point_plot):
+            return
+
+        # This is a heuristic approach until we think of how to hint this in a
+        # generalizable way.
+        if stream_name == self.dim_stream:
+            dim_fields = self.dim_fields
+        else:
+            dim_fields = ['time']  # 'time' once LivePlot can do that
+
+        # Create a figure or reuse an existing one.
+
+        fig_name = '{} vs {}'.format(' '.join(sorted(columns)),
+                                     ' '.join(sorted(dim_fields)))
+        if self.overplot and len(dim_fields) == 1:
+            # If any open figure matches 'figname {number}', use it. If there
+            # are multiple, the most recently touched one will be used.
+            pat1 = re.compile('^' + fig_name + '$')
+            pat2 = re.compile('^' + fig_name + r' \d+$')
+            for label in plt.get_figlabels():
+                if pat1.match(label) or pat2.match(label):
+                    fig_name = label
+                    break
+        else:
+            if plt.fignum_exists(fig_name):
+                # Generate a unique name by appending a number.
+                for number in itertools.count(2):
+                    new_name = '{} {}'.format(fig_name, number)
+                    if not plt.fignum_exists(new_name):
+                        fig_name = new_name
+                        break
+        ndims = len(dim_fields)
```

Comment:
Should we be interested in the shape of the dims as well?

Reply:
I believe that each of these corresponds to a single axis and is therefore always 1D.

Reply:
I'm not certain if that assumption will hold up when we have spectra. A spectra as a series of temperature readings would have temperature and energy hinted as independent variables, which I think would show up here in `dim_fields`.